pandas/core/internals/managers.py

```diff
@@ -847,7 +847,7 @@ def _interleave(self, dtype=None, na_value=lib.no_default) -> np.ndarray:
         # Give EAs some input on what happens here. Sparse needs this.
         if isinstance(dtype, SparseDtype):
             dtype = dtype.subtype
-        elif is_extension_array_dtype(dtype):
+        elif is_extension_array_dtype(dtype) or is_dtype_equal(dtype, str):
```

Comment:
are you sure this is actually hit?

Reply:
Yes, this is to avoid initializing result as np.empty(dtype=str) which was creating an array with dtype "<U1" and then breaking things.

Before the change that caused this regression the dtype was always being set to the inferred dtype from _interleaved_dtype (object in this case), so here I'm trying to make sure that this still happens.

Reply:
make this a separate elif as it is *very* confusing here the way it is written